src/common/actions/snap-builds.js

```diff
@@ -18,27 +20,38 @@ const REQUEST_POST_OPTIONS = {
   credentials: 'same-origin'
 };
 
-export function fetchBuildsSuccess(builds) {
+export function fetchBuildsSuccess(id, builds) {
   return {
     type: FETCH_BUILDS_SUCCESS,
-    payload: builds
+    payload: {
+      id,
+      builds
+    }
   };
 }
 
-export function fetchBuildsError(error) {
+export function fetchBuildsError(id, error) {
   return {
     type: FETCH_BUILDS_ERROR,
-    payload: error,
+    payload: {
+      id,
+      error
+    },
     error: true
   };
 }
 
 // Fetch snap info (self_link) for given repository
 export function fetchSnap(repositoryUrl) {
+  const { fullName } = parseGitHubRepoUrl(repositoryUrl);
+
   return (dispatch) => {
     if (repositoryUrl) {
       dispatch({
-        type: FETCH_BUILDS
+        type: FETCH_BUILDS,
+        payload: {
+          id: fullName
+        }
```

Comment:
It doesn't clear the builds. It only changes `isFetching` to true.

```js
     case ActionTypes.FETCH_BUILDS:
        return {
          ...state,
         [payload.id]: {
           ...initialStatus,
           ...state[payload.id], // take previous status (keeping stored builds)
           isFetching: true
         }
        };
```

We don't clear builds before fetching, because on builds listing we fetch every 15 seconds, so we don't want them to disappear during loading.